controllers/welcome.rb

```diff
@@ -1,3 +1,89 @@
+# ===== welcome
 get '/' do
+  @apartments = Apartment.all
   erb :welcome
 end
+
+# ===== ===== ===== APARTMENTS ===== ===== =====
+# ===== show apartment
+get '/apartments/:id' do
+  @apartment = Apartment.find(@params[:id])
+  erb :apartment/show
+end
+
+# ===== edit apartment
+get '/apartments/:id/edit' do
+  @apartment = Apartment.find(@params[:id])
+  erb :apartment/edit
+end
+
+# ===== update apartment
+put '/apartments/:id/update' do
+  @apartment = Apartment.find(@params[:id])
+  @apartment.update(address: params[:address], monthly_rent: params[:monthly_rent], sqft: params[:sqft], num_beds: params[:num_beds], num_baths: params[:num_baths])
+  redirect "/apartments"
+end
+
+# ===== new apartment
+get '/apartments/new' do
+  erb :apartment/new
+end
+
+# ===== create apartment
+post '/apartments/create' do
+  Apartment.create(address: params[:address], monthly_rent: params[:monthly_rent], sqft: params[:sqft], num_beds: params[:num_beds], num_baths: params[:num_baths])
```

Comment:
This approach to creating isn't wrong per se, but the later way we used in Tunr is generally preferred. By that, I mean where we 'namespace the form inputs, to look like:
`<input name="apartment[address]>`
instead of just
`<input name="address">`.

This way, we can just do the following in our controller:
`new apartment = Apartment.create(params[:apartment]`